File: 3751-4000/3838. Weighted Word Mapping/Solution.java
class Solution {
    public String mapWordWeights(String[] words, int[] weights) {
        int n = words.length ; 
        StringBuilder ans = new StringBuilder(); 
        for( int i=0 ; i<n ; ++i ){
                int curr = 0 ; 
                for( char c : words[i].toCharArray() ){
                    curr += weights[c-'a'] ; 
                }
                curr %= 26 ; 
                curr = 25 - curr ; 
                ans.append( (char) ( 'a' + curr ) ) ; 
        }
        return ans.toString() ; 
    }
}

File: 3751-4000/3838. Weighted Word Mapping/solution.cpp
#include <bits/stdtr1c++.h>

using namespace std ; 

class Solution {
public:
    string mapWordWeights(vector<string>& words, vector<int>& weights) {
        int n = words.size() ; 
        int m = weights.size() ; 
        string ans(n , '-') ; 
        for( int i=0 ; i<n ; ++i ){
            int score = 0 ; 
            for( int j=0 ; j<words[i].size() ; ++j ){
                score += weights[ words[i][j] - 'a' ] ; 
            }
            score %= 26 ; 
            ans[i] = ( char( 'z' - score ) ) ; 
        }
        return ans ;  
    }
};